lib/expression/callbacks/standard.ex

```diff
@@ -1053,55 +1059,124 @@ defmodule Expression.Callbacks.Standard do
         {nil, nil}
       end
 
-    %{"__value__" => !!date, "date" => date, "datetime" => datetime}
+    %{
+      "__value__" => !!(date || datetime),
+      "match" => datetime || date,
+      "date" => date,
+      "datetime" => datetime
+    }
   end
 
   @doc """
   Tests whether `expression` is a date equal to `date_string`
   """
   @expression_doc expression: "has_date_eq(\"the date is 15/01/2017\", \"2017-01-15\")",
-                  result: true
+                  result: %{
+                    "__value__" => true,
+                    "match" => ~D[2017-01-15],
+                    "test" => ~D[2017-01-15]
+                  }
   @expression_doc expression:
                     "has_date_eq(\"there is no date here, just a year 2017\", \"2017-01-15\")",
-                  result: false
+                  result: %{
+                    "error" => %{
+                      "__type__" => "expression/v1error",
+                      "error" => true,
+                      "message" => "The first argument is nil"
+                    },
+                    "__value__" => false,
+                    "match" => nil,
+                    "test" => ~D[2017-01-15]
+                  }
   def has_date_eq(ctx, expression, date_string) do
     [expression, date_string] = eval_args!([expression, date_string], ctx)
     found_date = DateHelpers.extract_dateish(expression)
     test_date = DateHelpers.extract_dateish(date_string)
-    # Future match result: found_date
-    found_date == test_date
+
+    with {:ok, match} <- date_compare(found_date, test_date) do
```

Comment:
I agree withe credo regarding not using with for a single check. case is sufficient.

Reply:
thanks!! fixing